tutorials/viewports/custom_postprocessing.rst

```diff
@@ -1,81 +1,74 @@
-render our :ref:`Viewport <class_Viewport>`. This can either be a 2D or 3D scene. The benefit 
-of a 2D scene is it is easy to set up a fullscreen :ref:`ViewportContainer <class_ViewportContainer>` 
-to display our :ref:`Viewport <class_Viewport>`. 
+    Rendering using a ``Viewport`` gives you control over the
+    how the scene render, including the framerate, and you can use the
+    ``ViewportContainer`` to render 3D objects in a 2D scene.
 
-For this demo let's use a :ref:`Node2D <class_Node2D>` with a 
-:ref:`ViewportContainer <class_ViewportContainer>` and finally a :ref:`Viewport <class_Viewport>`. 
-
-Your hierarchy should look like this:
+For this demo we will use a :ref:`Node2D <class_Node2D>` with a ``ViewportContainer`` and finally a
+``Viewport``. Your Node tab should look like this:
 
 .. image:: img/post_hierarchy1.png
 
-Inside the :ref:`Viewport <class_Viewport>` you can have whatever you want. This will contain 
-your main scene. For this tutorial we will use a field of random boxes.
+Inside the ``Viewport`` you can have whatever you want. This will contain
+your main scene. For this tutorial we will use a field of random boxes:
 
 .. image:: img/post_boxes.png
 
-In order to take advantage of the :ref:`Viewport <class_Viewport>`, attach a material to the 
-:ref:`ViewportContainer <class_ViewportContainer>`. 
+Add a new a :ref:`ShaderMaterial <class_ShaderMaterial>` to the ``ViewportContainer``, and assign a new
+shader resource to it. You can access your rendered ``Viewport`` with the built-in ``TEXTURE`` uniform.
 
-The benefit of using the :ref:`ViewportContainer <class_ViewportContainer>` is that the 
-:ref:`Viewport <class_Viewport>` can be accessed easily in the shader with the built-in 
-``TEXTURE`` uniform. You can choose not to use a :ref:`ViewportContainer <class_ViewportContainer>`, 
-but if you do so you will need to create your own uniform in the shader and pass the 
-:ref:`Viewport <class_Viewport>` texture in manually. To do so, add a uniform to the shader you use.
+.. note::
 
-::
+    You can choose not to use a ``ViewportContainer``, but if you do so you will
+    need to create your own uniform in the shader and pass the ``Viewport`` texture in
+    manually, like so:
 
-  //In Godot Shader
-  uniform sampler2D ViewportTexture;
+    ::
 
-And you can pass the texture into the shader from GDScript like so:
+      // Inside the Shader
+      uniform sampler2D ViewportTexture;
 
-::
+    And you can pass the texture into the shader from GDScript like so:
 
-  #In Gdscript
-  func _ready():
-    $Sprite.material.set_shader_param("ViewportTexture", $Viewport.get_texture())
+    ::
 
-Now, assuming you are using a :ref:`ViewportContainer <class_ViewportContainer>`, add a 
-:ref:`ShaderMaterial <class_ShaderMaterial>` to the :ref:`ViewportContainer <class_ViewportContainer>`
-and add the following code.
+      #In Gdscript
```

Comment:
Maybe put a space between the `#` and `In Gdscript`? Also capitalize `GDScript`.